tests/e2e/e2e_exec_test.go

```diff
@@ -292,6 +293,55 @@ func (s *IntegrationTestSuite) execBankSend(
 	s.executeGaiaTxCommand(ctx, c, gaiaCommand, valIdx, s.expectErrExecValidation(c, valIdx, expectErr))
 }
 
+type txBankSend struct {
+	from      string
+	to        string
+	amt       string
+	fees      string
+	log       string
+	expectErr bool
+}
+
+type successBankSend struct {
+	mu           sync.Mutex
+	successCount int
+}
+
+func (sc *successBankSend) inc() {
+	sc.mu.Lock()
+	defer sc.mu.Unlock()
+
+	sc.successCount++
+}
+
+func (s *IntegrationTestSuite) execBankSendBatch(
+	c *chain,
+	valIdx int,
+	txs []txBankSend,
+) int {
+	sucessBankSendCount := successBankSend{
+		successCount: 0,
+	}
+	var wg sync.WaitGroup
+
+	for i := 0; i < len(txs); i++ {
+		wg.Add(1)
+		go func(idx int) {
+			defer wg.Done()
+
+			s.T().Logf(txs[idx].log)
+			s.execBankSend(c, valIdx, txs[idx].from, txs[idx].to, txs[idx].amt, txs[idx].fees, txs[idx].expectErr)
+			if !txs[idx].expectErr {
+				sucessBankSendCount.inc()
+			}
+		}(i)
+
+		wg.Wait()
+	}
```

Comment:
the wait call should be call outside the loop, if not, the waitgroup will run synchronous

```suggestion
	}
	wg.Wait()
```